File: Medii de Proiectare si Programare/TrackAndField-C#/TrackAndField/repository/IRegistrationRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TrackAndField.model;

namespace TrackAndField.repository
{
    interface IRegistrationRepository : ICrudRepository<Tuple<Participant, Track>, Registration> 
    {
        /**
         * param: track
         *        ageGroup
         * return: all participants registered by track & age group and the number of registrations per participant
         */
        Dictionary<Participant, int> GetParticipantsByTrackAndAgeGroup(Track track, string ageGroup);
    }
}

File: Medii de Proiectare si Programare/TrackAndField-Java/src/main/java/repository/IRegistrationRepository.java
package repository;

import model.*;

import java.util.Map;

public interface IRegistrationRepository extends ICrudRepository<Tuple<Participant, Track>, Registration> {
    /**
     *
     * @param track
     * @param ageGroup
     * @return all participants registered by track & age group and the number of registrations per participant
     */
    Map<Participant, Integer> getParticipantsByTrackAndAgeGroup(Track track, String ageGroup);
}
